packages/smooth_app/lib/pages/product/edit_new_packagings_helper.dart

```diff
@@ -75,23 +75,36 @@ class EditNewPackagingsHelper {
 
     void addIfNotEmpty(final String text) {
       if (text.isNotEmpty) {
-        result.add(text);
+        result.add('$text ');
       }
     }
 
-    addIfNotEmpty(controllerUnits.text);
+
+    if (controllerUnits.text.isNotEmpty) {
+      result.add('${controllerUnits.text} x ');
+    }
+    
     addIfNotEmpty(controllerShape.text);
-    addIfNotEmpty(controllerMaterial.text);
-    addIfNotEmpty(controllerRecycling.text);
-    addIfNotEmpty(controllerWeight.text);
     addIfNotEmpty(controllerQuantity.text);
 
+    if (controllerMaterial.text.isNotEmpty & controllerWeight.text.isNotEmpty) {
+      result.add('(${controllerMaterial.text} : ${controllerWeight.text}g)');
+    }
+
+    else if (controllerMaterial.text.isNotEmpty ) { // Therefore controllerWeight.text is empty
+          result.add('(${controllerMaterial.text})');
+    }
+
```

Comment:
```suggestion
    if (controllerMaterial.text.isNotEmpty) {
      if (controllerWeight.text.isNotEmpty) {
        result.add('(${controllerMaterial.text} : ${controllerWeight.text}g)');
      } else {
        result.add('(${controllerMaterial.text})');
      }
    }
```

So you don't display the weight if the material is not specified, which is a bit strange.

Reply:
Yes, this is the same behavior as in the final product page. However, it might be interesting to have the mass in both cases (Nevertheless, the case where 'material' is empty and 'weight' is filled is quit rare).

Reply:
Again, I don't know where the changed display was suggested. And how rare the "the case where 'material' is empty and 'weight' is filled".
I would suggest at least to consider this very rare case with an additional test: "if weight is not empty and material is empty, let's display only weight".
By the way in English there's no space before a `:`. In French it's an unbreakable space.

Reply:
```suggestion
if (controllerMaterial.text.isNotEmpty & controllerWeight.text.isNotEmpty) {
      result.add('(${controllerMaterial.text}: ${controllerWeight.text}g)');
    }

    if (controllerMaterial.text.isNotEmpty & controllerWeight.text.isEmpty) {
          result.add('(${controllerMaterial.text})');
    }
  
    if (controllerMaterial.text.isEmpty & controllerWeight.text.isNotEmpty) {
          result.add('(${controllerWeight.text}g)');
    }
    // And nothing is added if they are both empty.
```

Thanks for your advises, here is a version taking this case into account. I am not sure the display change was suggested elsewhere.

Reply:
> By the way in English there's no space before a `:`. In French it's an unbreakable space.

@monsieurtanuki: on the server we have a translatable string 'sep' which is an empty string in English, and a space in French and other languages. Is this something already handled in some way in the app? I didn't see something equivalent in the lib/l10n/app_en.arb file

Reply:
@stephanegigandet We don't have this `sep` in Smoothie.
Could you list all languages that do have a space, e.g. taken from the server?

Reply:
@monsieurtanuki I just checked, we only do it for French.

```
=head2 separator_before_colon( $l )

=head3 Arguments

=head4 language code $l

In some languages like French, colons have a space before them.
e.g. "Valeur : 500" in French, "Value: 500" in English

This function returns a non-breaking space character for those languages.

=cut

sub separator_before_colon ($l) {

	if ($l eq 'fr') {
		return "\N{U+00A0}";
	}
	else {
		return '';
	}
}
```

Maybe we can put a similar function in Smoothie, it's likely to be useful in other places.